.github/workflows/auto-docs-preview.yml

```diff
@@ -0,0 +1,263 @@
+# GitHub Action workflow for automatically creating documentation preview links in PRs
+# This workflow addresses issue #3189 to automate the preview creation process
+
+name: Auto-generate Docs PR Preview
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+      - 'SECURITY.md'
+      - 'CODE_OF_CONDUCT.md'
+      - 'GOVERNANCE.md'
+      - 'ONBOARDING.md'
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+
+jobs:
+  generate-docs-preview:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - name: Check out code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8
+
+      - name: Generate Documentation Preview
+        uses: actions/github-script@ed597411d8f924073f98dfc5c65a23a2325f34cd
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prNumber = context.payload.pull_request.number;
+            const headRepo = context.payload.pull_request.head.repo;
+            const headBranch = context.payload.pull_request.head.ref;
+            const headOwner = context.payload.pull_request.head.repo.owner.login;
+            const headRepoName = context.payload.pull_request.head.repo.name;
+            const baseRepo = context.payload.repository.full_name;
+            
+            console.log(`Processing PR #${prNumber}`);
+            console.log(`Head repo: ${headOwner}/${headRepoName} (${headRepo.full_name})`);
+            console.log(`Base repo: ${baseRepo}`);
+            console.log(`Head branch: ${headBranch}`);
+            
+            // Check if this is from a fork - compare against the official kubestellar repo
+            const isFromFork = headRepo.full_name !== 'kubestellar/kubestellar';
+            console.log(`Is from fork: ${isFromFork}`);
+            
+            // Special case: Allow rishi-jat's repo for testing the automation
+            const isTestingRepo = headOwner === 'rishi-jat' && headRepoName === 'kubestellar';
+            
+            if (!isFromFork && !isTestingRepo) {
+              console.log('PR is not from a fork, skipping preview generation');
+              return;
+            }
+            
+            if (isTestingRepo) {
```

Comment:
The `if` here is superfluous, the `return` in the contrary case mean that this code is only reached when this condition is true.